pkg/types/types_test.go

```diff
@@ -215,7 +232,59 @@ func TestMergeManyMinorityMismatchedMetrics(t *testing.T) {
 		IsAbsent: []bool{false, false},
 	}
 
-	got, ps, ins := MergeMetrics(input, true, 10)
+	got, ps, ins, fixed := MergeMetrics(input, cfg.ReplicaMatchModeCheck, 10)
+	if len(got) != 1 {
+		t.Errorf("Expected 1 metric, got %d", len(got))
+	}
+
+	if !MetricsEqual(got[0], expected) {
+		t.Errorf("Merge failed\nExp: %+v\nGot: %+v\n", expected, got[0])
+	}
+
+	if ps != 2 {
+		t.Errorf("Expected 2 metric points, got %d", ps)
+	}
+	if ins != 1 {
+		t.Errorf("Expected 1 mismatched point, got %d", ins)
+	}
+
+	if fixed != 0 {
+		t.Errorf("Expected 0 fixed mismatch point , got %d", fixed)
+	}
+}
+
+func TestMergeManyMinorityMismatchedMetricsWithMajority(t *testing.T) {
+	input := [][]Metric{
```

Comment:
some of this `input` can be deduplicated amongst the [new] tests?

Reply:
There are a lot of things to improve in tests. It's very basic and naive at the moment. I didn't spend time on making it better because I thought dedicating a time to improve the whole testing in the project makes more sense. wdyt?

Reply:
ack